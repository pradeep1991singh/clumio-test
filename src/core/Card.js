import React from "react";

const Card = ({ title, posterPath, genreList, releaseYear }) => {
  return (
    <div className="max-w-sm overflow-hidden">
      <img className="w-full" src={posterPath} alt="Card" />
      <div className="px-6 py-4">
        <div className="font-bold text-xl mb-2 text-blue-500">{title}</div>
        <p className="text-gray-700 text-base dark:text-gray-500">
          {genreList}, {releaseYear}
        </p>
      </div>
    </div>
  );
};

export default Card;
