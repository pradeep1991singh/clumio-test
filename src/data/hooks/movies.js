import { useQuery } from "react-query";
import { filterList } from "../../services/utils";
import {
  getPopular,
  getTrending,
  getNewest,
  getTopRated,
} from "../api/movieApi";
import { useGetGenreList } from "./genres";

export const useGetPopularList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("popularList", getPopular, {
    select: (data) => filterList(data, filterType, filter, genres), // debounce this
    refetch: true,
  });
};

export const useGetTrendList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("trendList", getTrending, {
    select: (data) => filterList(data, filterType, filter, genres), // debounce this
    refetch: true,
  });
};

export const useGetNewestList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("newestList", getNewest, {
    select: (data) =>
      filterList({ results: [data] }, filterType, filter, genres), // debounce this
    refetch: true,
  });
};

export const useGetTopRatedList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("topRatedList", getTopRated, {
    select: (data) => filterList(data, filterType, filter, genres), // debounce this
    refetch: true,
  });
};
