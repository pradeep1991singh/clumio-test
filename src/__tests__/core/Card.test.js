import React from "react";
import { render } from "@testing-library/react";
import Card from "../../core/Card";
import { movieMock } from "../../__mocks__/movies.mock";

describe("<Card />", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<Card {...movieMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
