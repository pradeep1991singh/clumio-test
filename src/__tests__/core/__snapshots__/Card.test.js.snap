// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-sm overflow-hidden"
  >
    <img
      alt="Card"
      class="w-full"
      src="example.com"
    />
    <div
      class="px-6 py-4"
    >
      <div
        class="font-bold text-xl mb-2 text-blue-500"
      >
        title
      </div>
      <p
        class="text-gray-700 text-base dark:text-gray-500"
      >
        genre1genre2, 2022
      </p>
    </div>
  </div>
</DocumentFragment>
`;
